### PR TITLE
rule: allow usage of Label struct with Rule APIs

### PR DESCRIPTION
**Pull request by t-8ch**
**What type of PR is this?**

Feature

**What package or component does this PR mostly affect?**

rule

**What does this PR do? Why is it needed?**

The Label is a convenient structure for custom logic. Allow its use with the Rule APIs so the user can directly feed structured data back into Gazelle.

### CODE REVIEW

File: rule/value.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ package rule
 
 import (
 	"fmt"
+	"github.com/bazelbuild/bazel-gazelle/label"
```
Comment (fmeum) [inline rule/value.go:20]: I can't remember whether we have tooling for that, but please move this import down into the separate block below.

File: rule/value_test.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ limitations under the License.
 package rule
 
 import (
+	"github.com/bazelbuild/bazel-gazelle/label"
```
Comment (fmeum) [inline rule/value_test.go:19]: Also move this down.

**Review by fmeum — APPROVED** [review]

**Comment by t-8ch** [comment]
An alternative would be to introduce an interface to make the conversion pluggable.

**Comment by fmeum** [comment]
> An alternative would be to introduce an interface to make the conversion pluggable.

We could look into that if more types come up. This particular one seems clear enough since `Label(...)` isn't available in BUILD files.

**Comment by t-8ch** [comment]
> An alternative would be to introduce an interface to make the conversion pluggable.

Turns out there is already
```go
// BzlExprValue is implemented by types that have custom translations
// to Starlark values.
type BzlExprValue interface {
	BzlExpr() bzl.Expr
}
```

**Comment by fmeum** [comment]
Do you want to submit a follow-up PR that implements this interface for Label instead?

**Pull request merged**